cloudca/resource_cloudca.go

```diff
@@ -40,3 +45,22 @@ func readIntFromString(valStr string) int {
 	}
 	return valInt
 }
+
+// Provides a common, simple way to deal with 404s.
+func handleNotFoundError(err error, d *schema.ResourceData) error {
+	if ccaError, ok := err.(api.CcaErrorResponse); ok {
+		if ccaError.StatusCode == 404 {
+			fmt.Errorf("Entity (id=%s) not found", d.Id())
+			d.SetId("")
+			return nil
```

Comment:
The way our plugin (and [other plugins](https://github.com/hashicorp/terraform/blob/master/builtin/providers/digitalocean/resource_digitalocean_droplet.go#L520)) handle 404 is returning `nil` - it just handles remotely deleted entities.